lib/HALTarget/src/Network.cpp

```diff
@@ -316,13 +368,16 @@ void Network::unsubscribe(const String& topic)
 
 void Network::idleState()
 {
-    if ((false == m_clientId.isEmpty()) && (false == m_brokerAddress.isEmpty()) && (0U != m_brokerPort))
+    if ((false == m_clientId.isEmpty()) && (false == m_brokerAddress.isEmpty()) && (0U != m_brokerPort) &&
+        (false == m_wiFiSSID.isEmpty()))
     {
         (void)m_mqttClient.setServer(m_brokerAddress.c_str(), m_brokerPort);
         (void)m_mqttClient.setBufferSize(MAX_BUFFER_SIZE);
         (void)m_mqttClient.setKeepAlive(MQTT_KEEP_ALIVE_S);
         (void)m_mqttClient.setCallback([this](char* topic, uint8_t* payload, uint32_t length)
                                        { this->onMessageCallback(topic, payload, length); });

```

Comment:
Does all these functions have return values?

Reply:
Some return the client instance, allowing the function to be chained